base/abstractarray.jl

```diff
@@ -1094,6 +1094,8 @@ isempty(a::AbstractArray) = (length(a) == 0)
 
 
 ## range conversions ##
+AbstractVector{T}(r::AbstractRange) where {T} = map(T,r)
```

Comment:
Actually this breaks legitimate and currently working use cases such as `AbstractVector{Union{Missing, Int}}(1:10)`. This is the reason why StatsModels fails tests in the Nanosoldier run. We should continue returning a `Vector` when `T` differs from `eltype(r)`.